apollo-federation/src/query_graph/condition_resolver.rs

```diff
@@ -29,6 +43,7 @@ pub(crate) enum ConditionResolution {
     Satisfied {
         cost: QueryPlanCost,
         path_tree: Option<Arc<OpPathTree>>,
+        context_map: Option<IndexMap<String, ContextMapEntry>>,
```

Comment:
I was thinking it would be more performant in the > 99% of cases where there is no context set?

Reply:
More performant how? The default map is empty and has no allocation, so an empty map is trivial to construct. Although, I suppose having this be an option does communicate to future devs that this will not always be here.